docs/DeveloperGuide.md

```diff
@@ -102,7 +118,38 @@ If the isExitCommand() method returns true on the string representing user input
 method of the AllOnUs class, which then calls the static exit method in the same class to print a termination message, and then finally control
 is returned to the OS. 
 
-### Modules
+### Study Manager component
+API: `StudyManager.java`
+
+![](images/ModuleClassDiagram.png)
+
+
+The `StudyManager` component,
+1. Stores the academic schedule, i.e. all `Module` objects are contained in a `modulesList` object.
+2. The storage can be bound to `modulesList` list such that everytime a change is observed in the list it is saved on to a file.
+3. Does not depend on other components like `ExpenseTracker` and `ContactsManager`
+
+How the StudyManager works:
+1. Based on a command entered by the user, StudyManager can perform the following functions 
+   1. add 
+   2. list
+   3. delete
+   4. edit
+   5. find 
+   6. read ics
+2. There are two ways to add to the `moduleList` to keep track of academic schedule 
+   1. Either the user can choose to add the modules one by one. (See user guide for add feature)
+   2. Or the user can choose to read from NUSMods .ics calendar file of their academic schedule.  (See user guide for read from ics feature)
+      1. This is more convenient as it can add multiple modules at a single go.
+      2. This feature makes use of the `ModuleCalendarReader.java` API.
+3. If there are any errors in the entry of `Module` parameters they are handled by the various exceptions in the `exceptions` package.
```

Comment:
Just a stylistic choice, but perhaps bullet points would be nice here instead of numbering?

Reply:
Sounds good!